evalml/tests/automl_tests/test_automl.py

```diff
@@ -1463,6 +1464,8 @@ def new(self, parameters, random_seed=0):
 
         def clone(self):
             return self.__class__(self.parameters, random_seed=self.random_seed)
+    mock_score = MagicMock(side_effect=[{"Log Loss Binary": 1, "F1": val} for val in fold_scores])
+    DummyPipeline.score = mock_score
```

Comment:
Why is this change necassary? I thought the old mocking would work just fine.

Reply:
Previously, we separated out and mocked "evalml.pipelines.ModeBaselineBinaryPipeline.score" differently from DummyPipeline. Now, we no longer have ModeBaselineBinaryPipeline to mock, so I'm mocking the first pipeline (baseline) via the BinaryClassificationPipeline class and then mocking this custom class with this line.